lib/omnibus/semantic_version.rb

```diff
@@ -0,0 +1,53 @@
+#
+# Copyright 2015 Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'mixlib/versioning'
+
+module Omnibus
+  class SemanticVersion
+    def initialize(version_string)
+      @prefix = if version_string =~ /^v/
+                  "v"
+                else
+                  ""
+                end
+      @version = Mixlib::Versioning.parse(version_string.gsub(/^v/, ""))
```

Comment:
What happens if a version string does not adhere to a format `Mixlib::Versioning` can parse? Can we gracefully throw an exception and add some test coverage for that case?